### PR TITLE
API documentation

### PR DESCRIPTION
**Pull request by ironjan**
- Adds [play-swagger](https://github.com/iheartradio/play-swagger) to generate swagger.json
- Adds documentation for the users-API as defined in http://atlassian-hip.cs.upb.de:8080/browse/HIPCMS-314
- The documented API-endpoints currently return HTTP 501 Not Implemented
- The documentation does not yet contain request and response models

swagger.json: https://gist.github.com/ironjan/3fe246c434208f9e8344

screenshot at 2016-03-18 19 14 59

### CODE REVIEW

File: app/controllers/api/protocol/UserResponseModel.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    * Deserializer for java.util.UUID
+    * From http://alvinalexander.com/java/jwarehouse/play-framework-2.3/framework/src/play-json/src/main/scala/play/api/libs/json/Reads.scala.shtml
+    */
+  def uuidReader(checkUuuidValidity: Boolean = false): Reads[UUID] = new Reads[UUID] {
+
+    import java.util.UUID
+
+    import scala.util.Try
+
+    def check(s: String)(u: UUID): Boolean = (u != null && s == u.toString())
```
Comment (ironjan) [inline app/controllers/api/protocol/UserResponseModel.scala]: Codacy Issue found: [Avoid using null](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281812433/issues/source?bid=3144120&fileBranchId=3179406#l40)

File: app/controllers/yeoman/YeomanController.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/**
+  * This class is a delegator for com.tuplejump.playYeoman.Yeoman
+  * It needs to be used to prevent swagger-doc generation for yeoman.Routes
+  */
+class YeomanController {
+    import Yeoman._
+  
+      val index = Yeoman.index
+  
+      def at (file: String) = Yeoman.at(file)
```
Comment (ironjan) [inline app/controllers/yeoman/YeomanController.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281812436/issues/source?bid=3144120&fileBranchId=3179406#l15)

File: app/controllers/swagger/ApiSpecs.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import scala.concurrent.Future
+
+class ApiSpecs @Inject()(cached: Cached) extends Controller {
+    implicit val cl = getClass.getClassLoader
+
+      val domainPackage = "controllers.api.protocol"
+    private lazy val generator = SwaggerSpecGenerator(domainPackage)
+
+      def specs = cached("swaggerDef") {
```
Comment (ironjan) [inline app/controllers/swagger/ApiSpecs.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281812435/issues/source?bid=3144120&fileBranchId=3179406#l18)

File: app/controllers/api/UsersController.scala [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+  /**
+    *
+    * @param id
+    * @return
+    * @todo Add documentationt
+    * @todo Implement controller method
+    */
+  def updateUser(id: UUID) = play.mvc.Results.TODO
```
Comment (ironjan) [inline app/controllers/api/UsersController.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281812432/issues/source?bid=3144120&fileBranchId=3179406#l46)

File: app/controllers/api/UsersController.scala [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    * Finds the users matching the given filters.
+    *
+    * @param search The returned array will only contain users whose name or email address contains search.
+    *               If search is null, the users are neitehr filtered by name nor by email address.
+    * @param role   The returned array will only contain users whose role matches the given role. If role is null, the
+    *               users are not filtered by role.
+    * @return the users matching the given filter parameters.
+    * @todo Implement controller method
+    */
+  def userList(search: Option[String], role: Option[String]) = SecuredAction.async { implicit request =>
```
Comment (ironjan) [inline app/controllers/api/UsersController.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281812432/issues/source?bid=3144120&fileBranchId=3179406#l33)

File: app/controllers/api/protocol/UserResponseModel.scala (hunk 1 above)
Comment (ironjan) [inline app/controllers/api/protocol/UserResponseModel.scala]: Codacy Issue found: [Avoid using null](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281876692/issues/source?bid=3144120&fileBranchId=3179406#l40)

File: app/controllers/ApplicationController.scala [hunk 6]
```diff
@@ -27,14 +27,14 @@ class ApplicationController @Inject() (
    *
    * @return The result to display.
    */
-  def user = SecuredAction.async { implicit request =>
+    def user : Action[AnyContent] = SecuredAction.async { implicit request =>
```
Comment (uxxman) [inline app/controllers/ApplicationController.scala]: Can we use "Action[JsValue]" here like you used in other actions, to make things consistent? All return the same type of response.

Comment (ironjan) [inline app/controllers/ApplicationController.scala]: Research for conversion did not yield a result; it seems that Ok(…) always yields AnyContent, even if JsValue is passed in like here). Nonetheless I'll do some more research to find out if changing this is possible.

**Pull request merged**